AppCenter/AppCenter/Internals/Channel/MSACChannelGroupDefault.m

```diff
@@ -87,17 +87,13 @@ - (void)removeDelegate:(id<MSACChannelDelegate>)delegate {
 }
 
 - (void)enumerateDelegatesForSelector:(SEL)selector withBlock:(void (^)(id<MSACChannelDelegate> delegate))block {
-  NSArray *synchronizedDelegates;
-  @synchronized(self) {
-
-    // Don't execute the block while locking; it might be locking too and deadlock ourselves.
-    synchronizedDelegates = [self.delegates allObjects];
-  }
-  for (id<MSACChannelDelegate> delegate in synchronizedDelegates) {
-    if ([delegate respondsToSelector:selector]) {
-      block(delegate);
+    @synchronized(self.delegates) {
```

Comment:
You can't really do this. Check the comment. If `block(delegate)` calls this instance, it will end up with a deadlock.

Reply:
There is MSACDeadLockTests i mentioned in description